### PR TITLE
chore(docs): Update tutorial to Head API

### PR DESCRIPTION
**Pull request by marvinjude**
## Description

 - [x] Add some Head API intro to early steps in the tutorial
 - [x] Updates stepwise tutorial code snippets to use Gatsby Head API and not just add head tags directly in template
 - [x] Adds a `Seo` component task 
 - [x] Removes SEO tutorial https://www.gatsbyjs.com/tutorial/seo-and-social-sharing-cards-tutorial/
 - [x] Update tutorial example repo - https://github.com/gatsbyjs/tutorial-example/pull/6

### Documentation

https://www.gatsbyjs.com/docs/tutorial

[sc-53426]

### CODE REVIEW

File: docs/docs/tutorial/part-4/index.mdx [hunk 1]
```diff
@@ -420,8 +419,6 @@ const Layout = ({ pageTitle, children }) => {
 export default Layout
 ```
 
-![A screenshot of the home page in a web browser. The title of the page in the browser tab now says, "Home Page | My First Gatsby Site", and the top of the page has an unstyled paragraph that says, "My First Gatsby Site".](./index-page-with-site-title.png)
```
Comment (LekoArts) [inline docs/docs/tutorial/part-4/index.mdx:423]: Why did you remove the image?

Comment (marvinjude) [reply docs/docs/tutorial/part-4/index.mdx]: Ah! I think I was trying to reorder things at some point and didn't end up adding it back. Restored it now

File: docs/docs/tutorial/part-2/index.mdx [hunk 2]
```diff
@@ -15,6 +15,7 @@ To build out the basic page structure for your blog site, you'll need to know ab
 By the end of this part of the Tutorial, you will be able to:
 
 * Create **page components** to add new pages to your site.
+* Add title to your site using the **Gatsby Head API**
```
Comment (LekoArts) [inline docs/docs/tutorial/part-2/index.mdx:18]:
```suggestion
* Add a title to your site using the **Gatsby Head API**
```

File: docs/docs/tutorial/part-2/index.mdx [hunk 3]
```diff
@@ -183,13 +184,15 @@ import * as React from 'react'
...
   return (
     <main>
-      <title>Home Page</title>
       <h1>Welcome to my Gatsby site!</h1>
       <p>I'm making this by following the Gatsby Tutorial.</p>
     </main>
   )
 }
 
+// Page title that shows up on the browser tab
```
Comment (LekoArts) [inline docs/docs/tutorial/part-2/index.mdx:193]:
```suggestion
// Page title that shows up on the browser tab and in the <head> element
```

File: docs/docs/tutorial/part-2/index.mdx [hunk 4]
```diff
@@ -233,9 +235,31 @@ const AboutPage = () => {
 export default AboutPage
 ```
 
-2. In a web browser, visit `localhost:8000/about`. When your development server finishes rebuilding your site, the About page should look something like this:
+2. Add a page title: Gatsby lets you define a title and other [document head](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/head) related tags with the [Gatsby Head API](/docs/reference/built-in-components/gatsby-head/).
+When you export a `Head` component from your page file, It automatically picks all the head tags in your JSX and adds them to the [document head](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/head).
+This is important because it helps search engines like Google to better understand your site. For this tutorial we'll only be adding a title to this page but you can also add `<meta>`, `<links>` and other head related tags
```
Comment (LekoArts) [inline docs/docs/tutorial/part-2/index.mdx:240]:
```suggestion
2. Add a page title to your page. Gatsby lets you define a `<title>` and other [document metadata](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/head) with the [Gatsby Head API](/docs/reference/built-in-components/gatsby-head/). You have to export a component called `Head` from your page to apply the metadata.
Adding such metadata helps search engines like Google to better understand your site. For this tutorial we'll only be adding a title to this page but you can also later add other metadata.
```

File: docs/docs/tutorial/part-2/index.mdx [hunk 5]
```diff
@@ -233,9 +235,31 @@ const AboutPage = () => {
...
+  )
+}
+
+// Yay! We defined a page title. This shows up on the browser tab
+export const Head = () => <title>About Me<title/>
+
+export default AboutPage
+```
+
+3. In a web browser, visit `localhost:8000/about`. When your development server finishes rebuilding your site, the About page should look something like this:
```
Comment (LekoArts) [inline docs/docs/tutorial/part-2/index.mdx:260]: The "Key Gatsby Concept" below should also contain a mention of the Gatsby Head API

File: docs/docs/tutorial/part-2/index.mdx [hunk 6]
```diff
@@ -374,6 +398,8 @@ const AboutPage = () => {
   )
 }
 
+export const Head = () => <title>About Page</title>
```
Comment (LekoArts) [inline docs/docs/tutorial/part-2/index.mdx:401]:
```suggestion
export const Head = () => <title>About Me</title>
```

File: docs/docs/tutorial/part-2/index.mdx [hunk 7]
```diff
@@ -773,6 +796,7 @@ Take a moment to think back on what you've learned so far. Challenge yourself to
 
 * What's the difference between a page component and a building-block component?
 * How do you add a new page to your Gatsby site?
+* How do you add title to page?
```
Comment (LekoArts) [inline docs/docs/tutorial/part-2/index.mdx:799]:
```suggestion
* How do you add a title to a page?
```

File: docs/docs/tutorial/part-4/index.mdx [hunk 8]
```diff
@@ -523,6 +521,10 @@ const BlogPage = () => {
   )
 }
 
+export const Head = ({ data }) => (
+   <title> My Blog Posts | {data.site.siteMetadata.title}</title>
```
Comment (LekoArts) [inline docs/docs/tutorial/part-4/index.mdx:525]:
```suggestion
   <title>My Blog Posts | {data.site.siteMetadata.title}</title>
```

File: docs/docs/tutorial/part-6/index.mdx [hunk 9]
```diff
@@ -428,6 +436,12 @@ The JSON object in the Query Variables section should look something like the on
       }
     `
 
+    export const Head = ({ data }) => {
```
Comment (LekoArts) [inline docs/docs/tutorial/part-6/index.mdx:439]: This could be simplified to one liner

File: docs/docs/tutorial/part-6/index.mdx [hunk 10]
```diff
@@ -393,6 +397,10 @@ The JSON object in the Query Variables section should look something like the on
     `
     // highlight-end
 
+    export const Head = () => (
```
Comment (LekoArts) [inline docs/docs/tutorial/part-6/index.mdx:400]: This could be simplified to one liner

File: docs/docs/tutorial/part-6/index.mdx [hunk 11]
```diff
@@ -146,6 +146,10 @@ When you build your site, Gatsby looks at the page components in your `src/pages
       )
     }
 
+    export const Head = () => (
```
Comment (LekoArts) [inline docs/docs/tutorial/part-6/index.mdx:149]: This could be simplified to one liner

File: docs/docs/tutorial/part-5/index.mdx [hunk 12]
```diff
@@ -557,6 +557,10 @@ const BlogPage = ({ data }) => {
   )
 }
 
+const Head = () => (
```
Comment (LekoArts) [inline docs/docs/tutorial/part-5/index.mdx:560]: This could be simplified to one liner

File: docs/docs/tutorial/part-6/index.mdx [hunk 13]
```diff
@@ -477,6 +491,11 @@ The last step of Part 6 is to clean up your Blog page. Instead of rendering the
       }
     `
 
+    export const Head = () => (
```
Comment (LekoArts) [inline docs/docs/tutorial/part-6/index.mdx:494]: This could be simplified to one liner

File: docs/docs/tutorial/part-6/index.mdx [hunk 14]
```diff
@@ -524,6 +543,10 @@ The last step of Part 6 is to clean up your Blog page. Instead of rendering the
       }
     `
 
+     export const Head = () => (
```
Comment (LekoArts) [inline docs/docs/tutorial/part-6/index.mdx:546]: This could be simplified to one liner + formatting is off

File: docs/tutorial/seo-and-social-sharing-cards-tutorial/index.md [hunk 15]
```diff
@@ -4,7 +4,7 @@ title: Search Engine Optimization (SEO) and Social Sharing Cards with Gatsby
 
 Perhaps you've been approached by an SEO _expert_ who can maximize your revenue and page views by following these **Three Simple Tricks**! Relatively few people make the concerted effort to implement SEO in their web app. This tutorial will share some of the ins and outs of SEO and how you can implement common SEO patterns in your Gatsby web app, today. By the end of this post you'll know how to do the following:
 
-- Implement SEO patterns with [react-helmet][react-helmet]
+- Implement SEO patterns with the [Gatsby Head API](/reference/built-in-components/gatsby-head/)
```
Comment (LekoArts) [inline docs/tutorial/seo-and-social-sharing-cards-tutorial/index.md:7]: This whole tutorial should be deleted since we already have https://www.gatsbyjs.com/docs/how-to/adding-common-features/adding-seo-component/

**Review by LekoArts — CHANGES_REQUESTED** [review]
You'll also want to put up a PR updating the example https://github.com/gatsbyjs/tutorial-example

File: docs/tutorial/seo-and-social-sharing-cards-tutorial/index.md (hunk 15 above)
Comment (marvinjude) [reply docs/tutorial/seo-and-social-sharing-cards-tutorial/index.md:7]: The only difference is that this one approaches it from a "social cards" perspective, so I thought not to remove it.

If you feeling strongly about removing it then we should

File: docs/docs/tutorial/part-2/index.mdx (hunk 4 above)
Comment (marvinjude) [reply docs/docs/tutorial/part-2/index.mdx:240]: Shorter, I like it 👍

File: docs/docs/tutorial/part-2/index.mdx (hunk 5 above)
Comment (marvinjude) [reply docs/docs/tutorial/part-2/index.mdx:259]: Each of the key concepts down there (from here https://www.gatsbyjs.com/docs/tutorial/part-2/#use-the-link-component) seem to be talking about specific things like Link, Layout, and Stying, not sure where to slot that.

If we decide to create a new heading for the Head API then we'd have to elaborate and also stay consistent by adding a collapsible like  👇

Screenshot 2022-08-16 at 07 13 51

File: docs/docs/tutorial/part-2/index.mdx [hunk 16]
```diff
@@ -233,9 +235,30 @@ const AboutPage = () => {
 export default AboutPage
 ```
 
-2. In a web browser, visit `localhost:8000/about`. When your development server finishes rebuilding your site, the About page should look something like this:
+2. Add a page title to your page. Gatsby lets you define a `<title>` and other [document metadata](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/head) with the [Gatsby Head API](/docs/reference/built-in-components/gatsby-head/). You have to export a component called `Head` from your page to apply the metadata.
+Adding such metadata helps search engines like Google to better understand your site. For this tutorial we'll only be adding a title to this page but you can also later add other metadata.
```
Comment (LekoArts) [inline docs/docs/tutorial/part-2/index.mdx:239]:
```suggestion
Adding such metadata helps search engines like Google to better understand your site. For this tutorial you'll only be adding a title to this page but you can also later add other metadata.
```

File: docs/docs/tutorial/part-2/index.mdx [hunk 17]
```diff
@@ -233,9 +235,30 @@ const AboutPage = () => {
...
+const AboutPage = () => {
+  return (
+    <main>
+      <h1>About Me</h1>
+      <p>Hi there! I'm the proud creator of this site, which I built with Gatsby.</p>
+    </main>
+  )
+}
+
+// Yay! We defined a page title. This shows up on the browser tab
```
Comment (LekoArts) [inline docs/docs/tutorial/part-2/index.mdx:253]:
```suggestion
// Yay! You defined a page title. This shows up on the browser tab
```

File: docs/tutorial/seo-and-social-sharing-cards-tutorial/index.md (hunk 15 above)
Comment (LekoArts) [reply docs/tutorial/seo-and-social-sharing-cards-tutorial/index.md:7]: Yes, let's remove it

File: docs/docs/tutorial/part-2/index.mdx [hunk 18]
```diff
@@ -233,9 +235,30 @@ const AboutPage = () => {
...
+export const Head = () => <title>About Me<title/>
+
+export default AboutPage
+```
```
Comment (LekoArts) [inline docs/docs/tutorial/part-2/index.mdx:257]: Add a "Key Gatsby Concept" below this second point here about the Head API

**Review by LekoArts — CHANGES_REQUESTED** [review]

File: docs/docs/tutorial/part-2/index.mdx (hunk 18 above)
Comment (marvinjude) [reply docs/docs/tutorial/part-2/index.mdx:257]: 👍

File: docs/docs/tutorial/part-4/index.mdx [hunk 19]
```diff
@@ -462,7 +461,6 @@ const Layout = ({ pageTitle, children }) => {
 
   return (
     <div className={container}>
-      <title>{pageTitle} | {data.site.siteMetadata.title}</title>
```
Comment (marvinjude) [inline docs/docs/tutorial/part-4/index.mdx:465]: Moving this to Head API means we lose access to `siteMetaData` 

To fix that, we'd need to introduce additional code to get `siteMetaData` ( useStaticQuery / Page Query)

Another thing we can do is to completely remove " | `{data.site.siteMetadata.title}`"

What would you suggest we do here

File: docs/docs/tutorial/part-2/index.mdx [hunk 20]
```diff
@@ -233,9 +235,38 @@ const AboutPage = () => {
...
+<Announcement style={{marginBottom: "1.5rem"}}>
+
+**Key Gatsby Concept** 💡
 
-![A screenshot of "localhost:8000/about" in a web browser. It has a heading that says, "About Me", and a paragraph that says, "Hi there! I'm the proud creator of this site, which I built with Gatsby."](./about-page.png)
+You just added a title to the About page using the [Gastby Head API](https://www.gatsbyjs.com/docs/reference/built-in-components/gatsby-head/). You'll notice that the title is now displayed on the browser tab.
+With this API, you can also add other metadata to your site. For example, you can add a description to your site that shows up in search engines like Google.
```
Comment (LekoArts) [inline docs/docs/tutorial/part-2/index.mdx:263]:
````suggestion
You can use the [Gatsby Head API](/docs/reference/built-in-components/gatsby-head/) by exporting a named function called `Head` in your pages and templates that are used to create pages. 

Be sure to capitalize `Head` and please note that exporting this named function inside a component won't add the metadata to the `<head>`. The above works because you're exporting `Head` in a page inside `src/pages`.

You can add any valid `<head>` tags inside the function and they'll be added to the page, for example:

```js
export const Head = () => (
  <>
    <title>About Me<title/>
    <meta name="description" content="Your description" />
  </>
)
```

After going through this tutorial, be sure to check out [Adding an SEO Component](/docs/how-to/adding-common-features/adding-seo-component/).
````

File: docs/docs/tutorial/part-4/index.mdx (hunk 19 above)
Comment (LekoArts) [reply docs/docs/tutorial/part-4/index.mdx:465]: Let's remove it from the `<title>` tag then. Otherwise it would complicate things too much

**Review by LekoArts — CHANGES_REQUESTED** [review]
Also note that we'll need to adjust the notes in part 2/part 4 I think to say that they can't follow the commits anymore but should check out the project, as said in https://github.com/gatsbyjs/tutorial-example/pull/6#discussion_r946512964

**Review by LekoArts — APPROVED** [review]
